src/TrackArt.cpp

```diff
@@ -322,6 +323,45 @@ std::optional<ClipTitle> DoDrawAudioTitle(
    return std::nullopt;
 }
 
+wxString GetPitchShiftText(double clipSemitoneShift)
+{
+   wxString pitchShiftText;
+   double absShift = std::abs(clipSemitoneShift);
+   if (absShift >= 0.01)
+   {
+      pitchShiftText = wxString::Format("%.2f", absShift);
+      while (pitchShiftText.EndsWith("0"))
+         pitchShiftText.RemoveLast();
+      if (pitchShiftText.EndsWith("."))
+         pitchShiftText.RemoveLast();
+   }
+   return pitchShiftText;
+}
+
+//! Returns the width of the drawn text and icon.
+int DrawPitchOrSpeedIconIfItFits(
+   wxDC& dc, const wxRect& titleRect, const wxString& clipTitle,
+   const wxBitmap& icon, const wxString& text, int right, int availableWidth)
+{
+   const auto textWidth = dc.GetTextExtent(text).GetWidth();
+
+   const auto iconHeight = icon.GetHeight();
+   const auto iconWidth = textWidth == 0 ? 0 : icon.GetWidth();
+   const auto rectWidth = iconWidth + dc.GetTextExtent(text).GetWidth();
+
+   const auto iconTop =
+      titleRect.GetTop() + (titleRect.GetHeight() - iconHeight) / 2;
+   const auto textTop = titleRect.GetTop();
+   const auto height = titleRect.GetHeight();
+
+   if (rectWidth == 0 || rectWidth > availableWidth)
+      return 0;
+
+   dc.DrawText(text, right - textWidth, textTop);
+   dc.DrawBitmap(icon, right - textWidth - iconWidth, iconTop);
```

Comment:
Should you always draw that bitmap, or not in case you reassigned 0 to iconWidth?

Reply:
Oh I see the early return did that, but I found it a bit non obvious